From b24424337b49a7bac75b2aa60801546b736380b6 Mon Sep 17 00:00:00 2001
From: Bert van Hall <bert.vanhall@avionic-design.de>
Date: Thu, 17 Nov 2016 13:52:58 +0100
Subject: [PATCH] use pregenerated man page

This avoids help2man trying to execute the cross-compiled binary to get
its --help output.

Signed-off-by: Bert van Hall <bert.vanhall@avionic-design.de>
---
 Makefile.am    |  5 ++---
 nodm.8_shipped | 50 ++++++++++++++++++++++++++++++++++++++++++++++++++
 2 files changed, 52 insertions(+), 3 deletions(-)
 create mode 100644 nodm.8_shipped

diff --git a/Makefile.am b/Makefile.am
index 28709f9..eaeaaba 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -39,9 +39,8 @@ nodm_SOURCES = $(libsources)		\
 
 nodm_CFLAGS = -DNODM_SESSION='"$(sbindir)/nodm"'
 
-nodm.8: nodm
-	help2man --section=8 --name="X display manager for automatic logins" \
-		 --no-info --include=nodm-man-extras ./$< > $@
+nodm.8: nodm.8_shipped
+	cp $< $@
 
 man_MANS = nodm.8 \
            $(NULL)
diff --git a/nodm.8_shipped b/nodm.8_shipped
new file mode 100644
index 0000000..25ebab6
--- /dev/null
+++ b/nodm.8_shipped
@@ -0,0 +1,50 @@
+.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.47.4.
+.TH NODM "8" "November 2016" "nodm version 0.12" "System Administration Utilities"
+.SH NAME
+nodm \- X display manager for automatic logins
+.SH SYNOPSIS
+.B nodm
+[\fI\,options\/\fR]
+.SH DESCRIPTION
+\fBnodm\fP is a minimal display manager that simply logs in as a given user and
+starts an X session, without asking for username or password.
+.P
+On a normal computer, using \fBnodm\fP is a big security issue because it would
+give anyone access to the computer.
+.P
+However, there are cases where automatic login is needed: for example in an
+embedded system such as a mobile phone, or in a kiosk setup, or in a control
+panel for industrial machinery.  For those cases, \fBnodm\fP is simple to
+setup, lightweight, and it should do exactly the right thing.
+.SH OPTIONS
+.TP
+\fB\-\-help\fR
+print this help message
+.TP
+\fB\-\-version\fR
+print nodm's version number
+.TP
+\fB\-\-verbose\fR
+verbose outpout or logging
+.TP
+\fB\-\-quiet\fR
+only log warnings and errors
+.TP
+\fB\-\-nested\fR
+run a nested X server, does not require root.
+The server defaults to "/usr/bin/Xnest :1",
+override with NODM_X_OPTIONS
+.TP
+\fB\-\-[no\-]syslog\fR
+enable/disable logging to syslog
+.TP
+\fB\-\-[no\-]stderr\fR
+enable/disable logging to stderr
+.SH AUTHOR
+\fBnodm\fP is written and maintained by Enrico Zini <enrico@enricozini.org> and
+Joachim Breitner <nomeata@debian.org>.
+.SH "SEE ALSO"
+.BR xinit (1),
+.BR xdm (1),
+.BR gdm (1),
+.BR kdm (1).
-- 
2.7.4

